internal/pkg/addon/storage.go

```diff
@@ -106,3 +104,8 @@ func NewS3(input *S3Props) *S3 {
 		parser: template.New(),
 	}
 }
+
+var storageTemplateFunctions = map[string]interface{}{
```

Comment:
nit: would you mind moving the global variable to the top of the file? I don't know if there is official guidance on it but I think they tend to be written that way